### ISSUE TITLE
How to share types in monorepo

### ISSUE THREAD

**Issue by Gupcheg** [smapiot/piral#729]
How to share types across a monorepo so I can get types in different microfronts while shortening the import path.

For example, right now my DateFormats export comes from "`modules\evo.core\apps\evo.core.host-app\src\shared\types.ts`".

And I import it into another microfront like this:
`import { DateFormats } from '@evo.core/host-app/src/shared'`

Can I somehow shorten my import path to `@evo.core/host-app`?

Considering that I added the following lines to my tsconfig of the microfront where I want to import these types:
```
"@evo.core/host-app": ["./modules/evo.core/apps/evo.core.host-app/src/shared/"],
"@evo.core/host-app/*": ["./modules/evo.core/apps/evo.core.host-app/src/shared/*"],
```

Thanks in advance for your answer!

**Comment by FlorianRappl** [comment]
Yeah you can do that. Just use the `types` field in the *package.json*. It's a standard field used by TypeScript to know what should be the "main" / "root-level exports" from a package.

--> Set `types` to `./src/shared.ts` (or `./src/shared/index.ts` - whatever you use).

Hope that helps!

**Comment by Gupcheg** [comment]
Thank you very much! It helped
